mysql-test/main/tlsv13.test

```diff
@@ -0,0 +1,46 @@
+--source include/have_ssl_communication.inc
+--source include/have_tlsv13.inc
+
+#
+# BUG - SSL_CIPHER SYSTEM VARIABLE CANNOT CONFIGURE TLSV1.3 AND TLSV1.2 CIPHERS AT THE SAME TIME
+#
+# If users specify TLSv1.3 and TLSv1.2 ciphers, it will only configure the
+# TLSv1.3 ciphers correctly but then it will keep all TLSv1.2 ciphers enabled.
+#
+# This is a potential security vulnerability because users trying to restrict
+# secure TLSv1.3 and TLSv1.2 ciphers will unintentionally have insecure TLSv1.2
+# ciphers enabled on the database.
+#
+let $restart_parameters=--ssl-cipher=TLS_AES_128_GCM_SHA256:ECDHE-RSA-AES256-GCM-SHA384;
+source include/restart_mysqld.inc;
+# The builder, amd64-ubuntu-2004-debug, doesn't allow configuring TLSv1.3 cipher suites
```

Comment:
I think this comment is misleading. And the SQL below, too. I can't figure out what this test is doing, however I see that ssl_cipher_list uses doublequotes, not singlequotes which one would expect.

Can you explain the intention? It still fails on amd64-ubuntu-2004-debug, and I'm sure why, but I see it uses some older OpenSSL 1.1.1f

Reply:
I was observing that the TLSv1.3 cipher suites were not being modified. I thought it might have been something specific with the environment of that particular worker, so this was an attempt at skipping the test on that worker.

> but I see it uses some older OpenSSL 1.1.1f

Thanks for the info. I tested with MariaDB on OpenSSL 1.1.1.
It seems that on OpenSSL 1.1.1, if `ssl_cipher` includes a TLSv1.2 cipher in the list passed to `SSL_CTX_set_ciphersuites()`, it will result in an error and won't modify the available TLSv1.3 cipher list for the context.

So with OpenSSL 1.1.1, setting `--ssl-cipher=TLS_AES_128_GCM_SHA256:ECDHE-RSA-AES256-GCM-SHA384` will error on the call to `SSL_CTX_set_ciphersuites()` but succeed on the call to `SSL_CTX_set_cipher_list()`.
Thus the resulting `Ssl_cipher_list` will be `TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256:TLS_AES_128_GCM_SHA256:ECDHE-RSA-AES256-GCM-SHA384` instead of `TLS_AES_128_GCM_SHA256:ECDHE-RSA-AES256-GCM-SHA384`.